src/RF24StreamLogAppender.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * RF24StreamtLogAppender.h
+ *
+ *  Created on: 3 paź 2020
+ *      Author: wmarkowski
+ */
+
+#ifndef SRC_RF24STREAMLOGAPPENDER_H_
+#define SRC_RF24STREAMLOGAPPENDER_H_

```

Comment:
Maybe instead of header guards the `#pragma once` would be more reasonable this decade? I don't think anyone is using a compiler that old that it couldn't be used.

Reply:
Actually this is how I define all my headers. I thought pragma once was employed by default in compiler.

Reply:
`#pragma once` is not a default. Yes, `#ifndef` header guards work, but `#pragma once` is just a bit nicer.

Reply:
What do we do with this?
I have found this question: https://arduino.stackexchange.com/questions/66104/include-guards-vs-pragma-once

The given answer says that **#pragma once** guards on the absolute path level of the file, while **include guards** work on the content of the file. If the library is included twice from two different folders (which may in general happen, but it seem to never take a place in Arduino world) then **#pragma once** may not work.

Reply:
I like the way it is, but I'm not used to seeing the folder name also (just the filename).

Reply:
I'm used to have also those **include guards**. I'm developing in Eclipse CDT (Sloeber edition for Arduino) and it automatically puts the folder name as well. Theoretically you could have the same header file name in different folders.

Reply:
>Theoretically you could have the same header file name in different folders.

Yeah that stackoverflow answer you linked also makes a similar claim. I should probably also switch to using the folder names.

@Avamander would changing the rest of the RF24* libs to using a src folder be detrimental? I'm thinking more in terms of CMake. Please discount the doxygen implications; I can manage that with my eyes closed.

Reply:
> Theoretically you could have the same header file name in different folders.

This is usually not a problem.

'> would changing the rest of the RF24* libs to using a src folder be detrimental? I'm thinking more in terms of CMake.

In terms of CMake, I don't think so, but Arduino might be a bit pickier.

Reply:
[link from Arduino docs about library structure](https://arduino.github.io/arduino-cli/latest/library-specification/#layout-of-folders-and-files)

Reply:
If I'm not mistaken, the page says that as well, it'd mean deprecating old Arduino IDE support. Should be fine?